tests/functional/test_integration.py

```diff
@@ -1,14 +1,69 @@
 import pytest
+import requests
+import subprocess
+import time
+import yaml
 from ops.model import ActiveStatus
 from pytest_operator.plugin import OpsTest
 
 
 @pytest.mark.abort_on_fail
-async def test_build_and_deploy(ops_test: OpsTest, gunicorn_image, statsd_exporter_image):
+async def test_build_and_deploy_with_psql(
+    ops_test: OpsTest,
+    gunicorn_image,
+    statsd_exporter_image,
+    influx_model_name,
+):
+    result = subprocess.check_output(["juju", "controllers", "--format", "yaml"])
```

Comment:
It is possible to run juju commands using OpsTest using `OpsTest.juju`. Also, please check and see whether this provides the functionality that is needed: https://pythonlibjuju.readthedocs.io/en/latest/api/juju.controller.html

Also applicable to the remaining juju calls below

Reply:
This does not provides some functionalities I need, so I'll replace whenever I can, but as I'm deploying on two controllers at the same time and opsTest lacks cross-controller support, I doubt I'll manage to use opstest.juju for all the commands, specially the commands for the machine controller. Does that make sense?

Reply:
Were you able to convert this to `OpsTest.juju`?